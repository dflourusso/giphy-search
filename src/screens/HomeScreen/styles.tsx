import { StyleSheet } from 'react-native'
import styled from 'styled-components/native'
import Colors from '../../constants/Colors'

export const Container = styled.View`
  flex: 1;
`

export const Title = styled.Text`
  font-size: 20px;
  font-weight: bold;
`

export const Header = styled.View`
  flex-direction: row;
  padding: 8px;
`

export const SearchInput = styled.TextInput.attrs(() => ({
  placeholder: 'Search for gifs...',
}))`
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: ${Colors.backgroundInput};
  border-radius: 8px;
  margin: 4px;
  flex: 1;
`

export const styles = StyleSheet.create({
  contentContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
})
