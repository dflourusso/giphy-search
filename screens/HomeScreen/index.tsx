import * as React from 'react'
import { Container, Title } from './styles'

const HomeScreen: React.FC = () => {
  return (
    <Container>
      <Title>Home Screen</Title>
    </Container>
  )
}

export default HomeScreen
