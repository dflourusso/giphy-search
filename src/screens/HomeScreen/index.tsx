import { NavigationProp, useNavigation } from '@react-navigation/native'
import * as React from 'react'
import { FlatList } from 'react-native'
import { useDispatch, useSelector } from 'react-redux'
import { RootStackParamList } from '../../../types'
import GifThumb from '../../components/GifThumb'
import HistoryLink from '../../components/HistoryLink'
import SearchButton from '../../components/SearchButton'
import { RootState } from '../../store'
import { gifsActions } from '../../store/ducks/gifs'
import { Container, Header, SearchInput } from './styles'

const NUMBER_OF_COLUMNS = 3

const HomeScreen: React.FC = () => {
  const navigation = useNavigation<NavigationProp<RootStackParamList>>()
  const dispatch = useDispatch()
  const { data, search } = useSelector((state: RootState) => state.gifsReducer)

  const onSearch = React.useCallback(() => {
    dispatch(gifsActions.setPage(1))
    dispatch(gifsActions.requestData())
  }, [dispatch])

  const onChangeText = React.useCallback(
    (text: string) => {
      dispatch(gifsActions.setSearch(text))
    },
    [dispatch],
  )

  const addPage = () => {
    dispatch(gifsActions.addPage())
  }

  return (
    <Container>
      <Header>
        <SearchInput value={search} onChangeText={onChangeText} />
        <SearchButton onPress={onSearch} />
        <HistoryLink onPress={() => navigation.navigate('HistoryScreen')} />
      </Header>
      <FlatList
        numColumns={NUMBER_OF_COLUMNS}
        data={data}
        keyExtractor={(p) => p.id}
        renderItem={({ item }) => (
          <GifThumb
            numberOfColumns={NUMBER_OF_COLUMNS}
            uri={item.images.original.url}
          />
        )}
        scrollEventThrottle={16}
        onEndReached={addPage}
      />
    </Container>
  )
}

export default HomeScreen
