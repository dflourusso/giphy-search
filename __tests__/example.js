import * as React from 'react'
import renderer from 'react-test-renderer'
import HomeScreen from '../screens/HomeScreen'

it('Should be true', () => {
  expect(true).toBe(true)
})

it(`Should renders HomeScreen correctly`, () => {
  const tree = renderer.create(<HomeScreen>Snapshot test!</HomeScreen>).toJSON()

  expect(tree).toMatchSnapshot()
})
