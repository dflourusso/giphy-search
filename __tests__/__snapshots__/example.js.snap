// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should renders HomeScreen correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Home Screen
  </Text>
</View>
`;
